### PR TITLE
[CI] codecov updates

### PR DESCRIPTION
**Pull request by christophfroehlich**
Similar to https://github.com/ros-controls/ros2_controllers/pull/804

- Update of the packages
- Exclude test folders from stats
- Add codecov badge to readme
- Remove the invalid `branches` key from the `workflow_displatch`

Do we want to activate codecov also for humble?

### CODE REVIEW

**Comment by christophfroehlich** [comment]
I don't know why `unload_on_kill` test should fail now.

File: .github/workflows/ci-coverage-build.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 name: Coverage Build
 on:
   workflow_dispatch:
-    branches:
-      - master
```
Comment (bmagyar) [inline .github/workflows/ci-coverage-build.yml:5]: why remove this? it's handy for me! :D 
does it disturb anything?

Comment (christophfroehlich) [reply .github/workflows/ci-coverage-build.yml:5]: my github linter in vscode tells me that it is not a valid key of workflow_dispatch ;) and I couldn't find it in the github docs. Isn't it that you can choose the branch for running the workflow anyways?

Comment (bmagyar) [reply .github/workflows/ci-coverage-build.yml:5]: pro'lly right, let's try removing it

**Review by bmagyar — APPROVED** [review]

File: .github/workflows/ci-coverage-build.yml (hunk 1 above)
Comment (christophfroehlich) [reply .github/workflows/ci-coverage-build.yml:5]: Can you run this workflow now manually for every branch?

**Pull request merged**